File: nxflix-agents-python/src/nxflix_agents/routers/health.py
"""Health check API route."""

from fastapi import APIRouter
from pydantic import BaseModel

from nxflix_agents.config import get_settings

router = APIRouter(tags=["health"])


class HealthResponse(BaseModel):
    """Health check response."""

    status: str
    version: str
    provider: str
    model: str
    opik_enabled: bool


@router.get("/api/health", response_model=HealthResponse)
async def health_check():
    """Health check endpoint."""
    settings = get_settings()

    return HealthResponse(
        status="healthy",
        version="0.1.0",
        provider=settings.default_provider,
        model=settings.default_model,
        opik_enabled=settings.opik_enabled,
    )

File: nxflix-agents-ts/src/routers/health.ts
import { Router, Request, Response } from 'express';
import { settings } from '../config.js';

const healthRouter = Router();

// GET /api/health
healthRouter.get('/health', (_req: Request, res: Response) => {
  res.json({
    status: 'healthy',
    version: '0.1.0',
    provider: settings.defaultProvider,
    model: settings.defaultModel,
    opikEnabled: settings.opikEnabled,
  });
});

export { healthRouter };
